composer/utils/checkpoint.py

```diff
@@ -413,9 +413,13 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
             Raises:
                 ValueError if the data file is invalid.
             """
+            already_read = set()
```

Comment:
nit: rename to `validated_checkpoint_paths`

Reply:
done